crates/weaver_resolved_schema/src/any_value.rs

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! Specification of a resolved `AnyValue`.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::attribute::{EnumEntriesSpec, Examples, RequirementLevel};
+use weaver_semconv::stability::Stability;
+
+/// An `AnyValue` definition.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct AnyValue {
+    /// AnyValue name.
+    pub name: String,
+    /// Either a string literal denoting the type as a primitive or an
+    /// array type, a template type or an enum definition.
+    pub r#type: String,
+    /// A description of the type of the AnyValue
+    /// e.g. "string", "string[]", "int", "enum<enum_id>", "map<id>{ int, string }"
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub type_display: Option<String>,
```

Comment:
How is this different than `name` in the enum case?

For non-map descriptions, why is this needed?

For map-descriptions, should this be derived from the structure?

I'd prefer not to add this, find it tedious and remove it.  What's the rationale for having it?

Reply:
I agree, to the "Why is this needed". I accidently created this and it is an "interesting" short descriptive representation of the nested definition. And this is one of the major discussion topics about whether we do or do not want this (it's not really needed)

Reply:
And actually, thinking about this is a different way, it should be possible to "process" this in any template implementation (if required) and re-create this exact same thing by parsing the structure (might need some minor "helpers" to do things like "get" the nested object or "has" nested objects. But we can grow those later (if required).

Really, I found this as a interesting representation more from a documentation representation (for something like markdown) perspective rather than 'just" the tables.

Reply:
I agree having a descriptive type is interesting, but I don't think it should be user-input (especially given what you're showing it as examples). I think we can synthesize something that will be consistent.  We can provide that synthesized string to templates via  filter or function, it should NOT be in the model.

Please remove, and we can figure out how to expose it in a follow up PR.